### PR TITLE
Opening Component Browser by clicking on the output port

### PR DESCRIPTION
**Pull request by akavel**
### Pull Request Description

Double-clicking a node's output port or clicking the port with a right mouse button (RMB) creates a new node aligned to the clicked node. 

#### Visuals

The screencast below demonstrates the following features:
 - double-clicking the left mouse button on a node's output port;
 - clicking the right mouse button on a node's output port;
 - alignment of the nodes created as a result of the actions described above;
 - corner case: double-clicking (and RMB-clicking) on output ports of a "collapsed" ("enterable") node;
 - double-clicking on a "collapsed" ("enterable") node still allows entering the node when done over an area of the node that is not the node's output port;
 - basic support for nodes with multiple output ports (shown on the `interface` demo scene).

https://user-images.githubusercontent.com/273837/158991856-e0faa5f0-9d2f-44bd-bddd-ba314977db6e.mov

The supplementary screencast below demonstrates that double-clicking or RMB-clicking a node's output port cancels the action of dragging a new connection from a node.

https://user-images.githubusercontent.com/273837/158998097-100aed42-37ff-4467-939f-2b755ef0d3dc.mov

https://www.pivotaltracker.com/story/show/181076145

### Important Notes

 - The "double-clicking a node" shortcut was previously used to allow entering a "collapsed" node (for example, a node created by pressing the `cmd+g` keyboard shortcut after selecting a group of nodes). This PR keeps that functionality when the user double-clicks on a node, as long as the mouse is not positioned over the node's output ports.
 - The support for nodes with multiple output ports is currently very basic. The information about a port (`Crumb`) is passed into the `create_node` function, but it is not passed further to `NodeSource`. The Node Searcher currently does not support passing port information through `NodeSource`.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md), [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md), and [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH `./run dist` and `./run watch`.

### CODE REVIEW

File: app/gui/view/graph-editor/src/lib.rs
Comment (wdanilo) [inline app/gui/view/graph-editor/src/lib.rs:1391]: What is the reason to make this struct so heavyweight? Why do you need to keep `EdgeEndpoint` instead of `NodeId`, just like `DroppingEdge` does?

File: app/gui/view/graph-editor/src/lib.rs
Comment (farmaazon) [inline app/gui/view/graph-editor/src/lib.rs]: Perhaps `create_node` could take `way` by a reference?

**Review by wdanilo — CHANGES_REQUESTED** [review]

File: app/gui/view/graph-editor/src/lib.rs
Comment (farmaazon) [inline app/gui/view/graph-editor/src/lib.rs]: 1. Some of those lines are over 100 characters.
2. In this section, FRP nodes are for just creating the `way` structure. Please move nodes specific to creating node from clicking output to a separate block.

File: app/gui/view/graph-editor/src/lib.rs
Comment (farmaazon) [inline app/gui/view/graph-editor/src/lib.rs:494]: I would repeat or refer to information in `start_node_creation` that this is meant to be user interaction.

**Review by farmaazon — APPROVED** [review]

File: app/gui/view/graph-editor/src/lib.rs
Comment (farmaazon) [reply app/gui/view/graph-editor/src/lib.rs:1391]: It was by my request. If we want to support multiple output ports (and we want eventually), then the future implementor would already know where to start.

Comment (wdanilo) [reply app/gui/view/graph-editor/src/lib.rs:1391]: I am OK with adding the information. My question was why instead of adding simply `NodeId` we are adding the whole structure there.

**Review by wdanilo — APPROVED** [review]

**Comment by akavel** [comment]
(*Process note:* the code-review is now Approved, therefore the PR is waiting for an Acceptance/QA Review on Task https://www.pivotaltracker.com/story/show/181076145 before merge.)

<blockquote><div><strong><a href="https://www.pivotaltracker.com/story/show/181076145">Opening Component Browser by clicking on the output port - Pivotal Tracker</a></strong></div></blockquote>

File: app/gui/view/graph-editor/src/lib.rs
Comment (farmaazon) [reply app/gui/view/graph-editor/src/lib.rs:1391]: Because it makes sense in this structure. The node was created by clicking on the port - the fact that we don't make use of this information right now is an implementation detail. And I would not be worried with weight of crumbs: those `Vec`s are empty in most cases, and usually have no more than two elements otherwise.

**Review by vitvakatu — APPROVED** [review]
Accepted

**Pull request merged**
